Crypto/Crypto/Presentation/ViewModels/ForgotPasswordViewModel.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ForgotPasswordViewModel.swift
+//  Crypto
+//
+//  Created by Orlando Nicolas Marchioli on 03/08/2023.
+//
+
+import Foundation
+import Firebase
+
+class ForgotPasswordViewModel: ObservableObject{
+    @Published var alert: Bool
+    @Published var email: String
+    @Published var alertMessage: String
+    
+    init() {
+        self.alert = false
+        self.email = ""
+        self.alertMessage = ""
+    }
+    
+    func verify(){
+        var message = ""
+        
+        if self.email.isEmpty {
+            message = "You can't have an empty field"
+        } else if !self.email.contains("@") && !self.email.contains(".") {
+            message = "The email is not valid"
+        }
+        
+        if !message.isEmpty {
+            // Set the error message and alert status on the main thread
+            DispatchQueue.main.async {
```

Comment:
The same as the previous comment when we refactored the way that you are defining the state we will be able to avoid using DispatchQueue.main.async